### DISCUSSION TITLE
Added and Removed query filters.

### DISCUSSION THREAD

**Discussion by RicardRC** [GodotECS/godex#247] [Ideas & Proposals]
This way, we can clearly define operations on the three steps of the component lifetime:
On creation, on destruction, and during it's normal life (which can be further filtered using the Changed filter).
As with the changed filter, only do the internal stuff for components wich have any actual usage of the Added and Removed, respectively.

Would it be wise to remove a component that has been added from the Changed filter, or it would be better to keep added components on the results of changed filters?

**Comment by AndreaCatania** [comment]
> Would it be wise to remove a component that has been added from the Changed filter, or it would be better to keep added components on the results of changed filters?

Yes, you can remove a component that has been added to the changed filters. Check this: https://github.com/GodotECS/godex/blob/main/storage/dense_vector_storage.h#L44-L48

**Reply by RicardRC** [reply]
No, I meant if we should keep the current beaviour of added components being retrieved by changed filters or it would make more sense to completely distinguish between the two and added components no longer return on changed filters.

**Reply by AndreaCatania** [reply]
This is interesting. Distinguish the two things would be more natural, but probably less helpful. With the current behavior (the insert triggers `changed`) with one `System` we can always fetch and properly detect and propagate the change. If we split those, we will need two `System`s, to do the same work.
However, let's think about this, probably would make more sense split it.
